src/CommandingOfficers/Tech.java

```diff
@@ -223,23 +223,44 @@ public GameEventQueue getEvents(MapMaster gameMap)
       Set<XYCoord> dropLocs = new HashSet<XYCoord>();
       for( int i = 0; i < numDrops; ++i )
       {
-        TeleportEvent techDrop = generateDropEvent(gameMap, dropLocs, log);
+        TeleportAction techDrop = generateDropEvent(gameMap, dropLocs, log);
         if( null != techDrop )
         {
           // Store the new event.
-          dropLocs.add(techDrop.getEndPoint());
-          abilityEvents.add(techDrop);
+          dropLocs.add(techDrop.getMoveLocation());
+          abilityEvents.addAll(techDrop.getEvents(gameMap));
         }
         else
         {
           System.out.println("[TechdropAbility.getEvents] WARNING! Could not find drop location.");
         }
       }
 
+      // Figure out if this caused the enemy's defeat. The TeleportActions can't track
+      // this because multiple units may be smashed at once.
+      Map<Commander, Integer> smashes = new HashMap<Commander, Integer>();
+      for( XYCoord xyc : dropLocs )
+      {
+        Unit r = gameMap.getLocation(xyc).getResident();
+        if( null != r )
+        {
+          Integer oldVal = smashes.get(r.CO);
+          if( null == oldVal )
+            smashes.put(r.CO, 1);
+          else
+            smashes.put(r.CO, oldVal+1);
```

Comment:
```suggestion
          Integer oldVal = smashes.putIfAbsent(r.CO, 1);
          if( null != oldVal ) smashes.put(r.CO, oldVal+1);
```
Ayy, this one worked, I think